components/style/style_adjuster.rs

```diff
@@ -210,12 +207,13 @@ impl<'a, 'b: 'a> StyleAdjuster<'a, 'b> {
             return;
         }
 
-        if self.style.get_box().clone_display() == Display::Inline {
-            self.style.mutate_box().set_adjusted_display(Display::InlineBlock,
-                                                         false);
+        if let Some(box_) = self.style.get_box_if_mutated() {
+            // Save the display value without style adjustments now,
+            // which we need for hypothetical boxes.
+            let display = box_.clone_display();
+            box_.set_original_display(display);
```

Comment:
Same here.

Reply:
I think I need to follow the original condition to set the `original_display` here.